Kotlin arrayOfNulls() term entry : Issue #5310 (#5593)
Codecademy/docs: content/kotlin/concepts/arrays/terms/arrayOfNulls/arrayOfNulls.md

---
Title: 'arrayOfNulls()'
Description: 'Creates an array of a specified size with all elements initialized as null.'
Subjects:
  - 'Code Foundations'
  - 'Mobile Development'
Tags:
  - 'Android'
  - 'Arrays'
  - 'Map'
  - 'Kotlin'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> In Kotlin, the **`arrayOfNulls()`** function creates an array of a specified size with all elements initialized as `null`. This method is useful when you need to create an array of a specific size but don't yet have the values to initialize it with. This allows you to assign values at a later point.

Reviewer: ```suggestion
In Kotlin, the **`arrayOfNulls()`** function creates an array of a specified size with all elements initialized as `null`. This method is useful for creating an array with a defined size when the initial values are not yet available, allowing for value assignment at a later point.
```

## Syntax

```pseudo
fun <T> arrayOfNulls(size: Int): Array<T?>
```

> - `T`: The type of the elements in the array.

Reviewer: ```suggestion
- `T`: Represents the type of elements in the array.
```

- `size`: An integer specifying the size of the array to create.

> It returns an `Array<T?>` of the specified size, initialized with nulls.

Reviewer: ```suggestion
It returns an `Array<T?>` of the specified size, initialized with `null`.
```

## Example

The following example uses the `arrayOfNulls()` function:

```kotlin
fun main() {
    // Create an array of size 5 with all elements initialized as null
    val nullArray = arrayOfNulls<Int>(5)

    // Assign values to some elements in the array
    nullArray[0] = 2
    nullArray[1] = 4

    // Print the array after assigning values
    println(nullArray.contentToString())
}
```

> 

Reviewer: ```suggestion
## Output
```

This example results in the following output:

> 

Reviewer: Give line break before and after output block
```suggestion
## Output
```

> The code above generates the following output:

Reviewer: ```suggestion
This example results in the following output:
```
